### PR TITLE
Handle heterogenous MySQL execution environments

### PR DESCRIPTION
**Pull request by meatballhat**
with artisanally crafted PHP/HHVM goodies.

### CODE REVIEW

**Comment by meatballhat** [comment]
Appears to work https://staging.travis-ci.org/meatballhat/yolo-octo-adventure/builds/527158

File: lib/travis/build/appliances/services.rb [hunk 1]
```diff
@@ -24,6 +24,15 @@ def apply
               end
             end
 
+            if services.delete('mysql')
```
Comment (BanzaiMan) [inline lib/travis/build/appliances/services.rb]: Can we announce the start of `mysql` service? (or does it happen elsewhere?)

File: lib/travis/build/appliances/services.rb [hunk 2]
```diff
@@ -24,6 +24,15 @@ def apply
...
+              sh.if "$(grep -E '^socket = /run/mysql-5.6' #{HOME_DIR}/.my.cnf)" do
+                sh.cmd "sudo service mysql-5.6 start", assert: false, echo: true, timing: true
```
Comment (joepvd) [inline lib/travis/build/appliances/services.rb]: The service name change should be announced and documented for people who change the configuration. According to GitHub search, there are +/- 250 public repos who have `sudo service mysql` in the build config. Can the service be renamed/aliased?

File: lib/travis/build/templates/php.sh [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+        sed -i "/^[a-z_]*mysql.*socket/s:\$: ${MYSQL_UNIX_PORT}:" "${ini_file}"
+      done
+  fi
+
+  if [[ -f '<%= root %>/etc/hhvm/php.ini' ]]; then
+    echo "
+pdo_mysql.default_socket = ${MYSQL_UNIX_PORT}
+mysqli.default_socket = ${MYSQL_UNIX_PORT}
+hhvm.mysql.socket = ${MYSQL_UNIX_PORT}
+" | sudo tee -a '<%= root %>/etc/hhvm/php.ini' >/dev/null
```
Comment (joepvd) [inline lib/travis/build/templates/php.sh]: hhvm might need a restart after this config change.

**Comment by joepvd** [comment]
Notify when done:

https://secure.helpscout.net/conversation/274962408/38692/
https://github.com/travis-ci/travis-ci/issues/6855
https://github.com/travis-ci/travis-ci/issues/6842

File: lib/travis/build/appliances/services.rb (hunk 2 above)
Comment (meatballhat) [inline lib/travis/build/appliances/services.rb]: I suspect a service alias is the easiest solution.  I don't want to take on the burden of maintaining our own MySQL cookbook (again), and the community cookbook we're using adds a suffix to `mysql-` by default.

File: lib/travis/build/templates/php.sh (hunk 3 above)
Comment (meatballhat) [inline lib/travis/build/templates/php.sh]: SGTM

File: lib/travis/build/appliances/services.rb (hunk 2 above)
Comment (meatballhat) [inline lib/travis/build/appliances/services.rb]: The situation is worse than I thought: https://travis-ci.org/meatballhat/cloaked-octo-tyrion/builds/180440706

I'm going to work on cleaning up the various mysql service bits, including aliasing `mysql-5.6` => `mysql`, although I'm pretty sure that upstart is going to have opinions about this.

**Comment by meatballhat** [comment]
This should no longer be necessary thanks to https://github.com/travis-ci/travis-cookbooks/pull/784

**Pull request closed without merging**